src/Glimpse.Agent.Web/Broker/EndRequestMessage.cs

```diff
@@ -1,36 +1,33 @@
 ﻿using System;
-using System.Collections.Generic;
+using Glimpse.Common.Broker;
 using Microsoft.AspNet.Http;
 
 namespace Glimpse.Agent.Web
 {
-    public class EndRequestMessage : IMessageIndices
+    public class EndRequestMessage
     {
         public EndRequestMessage(HttpRequest request, Timing timing)
         {
+            // TODO: check if there is a better way of doing this
             Url = $"{request.Scheme}://{request.Host}{request.PathBase}{request.Path}{request.QueryString}";
             StartTime = timing.Start;
             EndTime = timing.End;
             Duration = timing.Elapsed.TotalMilliseconds;
             Method = request.Method;
-
-            Indices = new Dictionary<string, object> {
-                { "request.duration", Duration },
-                { "request.method", Method },
-                { "request.url", Url }
-            };
         }
 
-        public IReadOnlyDictionary<string, object> Indices { get; set; }
-
-        public double Duration { get; }
+        [PromoteToDuration]
+        public double? Duration { get; }
 
-        public DateTime StartTime { get; }
+        [PromoteToDateTime]
+        public DateTime? StartTime { get; }
```

Comment:
In this case, no. 

I changed it anyways since there isn't a way to signal an empty `DateTime` without it, and the `[PromoteToDateTime]` attribute can handle `null` cases directly (as opposed to `default(DateTime)`). My thinking here was more about who would look at this code as a pattern of usage in the future.